### PR TITLE
Deprecate info on initial states and polarized key

### PR DESCRIPTION
**Pull request by Radonirinaunimi**
From [v0.7.4](https://github.com/NNPDF/pineappl/releases/tag/v0.7.4), the `initial_state_` keys in pineappl is deprecated. This release also introduced the `convolution_type_` that specifies the type of convolution required by the initial state making the `polarized` key deprecated.

### CODE REVIEW

File: src/yadmark/benchmark/external/apfelpy_utils.py [hunk 1]
```diff
@@ -168,6 +168,8 @@ def tabulate_nc(ap, obs_name, sfobj, nq, qmin, qmax, thrs):
         tab_sf = ap.TabulateObjectD(sfobj[4].Evaluate, nq, qmin, qmax, 3, thrs)
     elif "bottom" in obs_name:
         tab_sf = ap.TabulateObjectD(sfobj[5].Evaluate, nq, qmin, qmax, 3, thrs)
+    else:
+        raise ValueError(f"'(total, light, charm, botton)' not found in {obs_name}!")
```
Comment (giacomomagni) [inline src/yadmark/benchmark/external/apfelpy_utils.py:172]:
```suggestion
        raise ValueError(f"'(total, light, charm, bottom)' not found in {obs_name}!")
```

File: src/yadmark/benchmark/external/apfelpy_utils.py [hunk 2]
```diff
@@ -246,6 +248,8 @@ def tabulate_cc(ap, obs_name, sfobj, nq, qmin, qmax, thrs):
             3,
             thrs,
         )
+    else:
+        raise ValueError(f"'(total, light, charm, botton)' not found in {obs_name}!")
```
Comment (giacomomagni) [inline src/yadmark/benchmark/external/apfelpy_utils.py:252]:
```suggestion
        raise ValueError(f"'(total, light, charm, bottom)' not found in {obs_name}!")
```

**Comment by giacomomagni** [comment]
Need to change also `convolute_with_one` with `convolve_with_one`, there are two occurrences in the docs.

**Comment by Radonirinaunimi** [comment]
> Need to change also `convolute_with_one` with `convolve_with_one`, there are two occurrences in the docs.

That's right, thanks! Addressed in [2fb8790](https://github.com/NNPDF/yadism/pull/289/commits/2fb87908ffea3251e9bc111184771197896ab447).

**Review by giacomomagni — APPROVED** [review]

**Review by felixhekhorn — APPROVED** [review]
Looks good to me! (I wished we had as many tests as in eko :innocent: )

**Comment by Radonirinaunimi** [comment]
> Looks good to me! (I wished we had as many tests as in eko 😇 )

Indeed! I guess this is still one of the predicaments of Yadism.

**Comment by Radonirinaunimi** [comment]
Actually, while checking that everything works fine until the generation of FK tables I realized that `convolve_with_one` although already merged in master is not part of `v0.7.4`, so I undid the changes.

I will write a test that at least checks this before merging!

**Comment by Radonirinaunimi** [comment]
I changed my mind though. The test I was thinking of should not belong into `yadism` so I'd merge this as it is.

**Pull request merged**